tests/sp.test/sp.sh

```diff
@@ -344,7 +344,9 @@ local function main(t)
   th1:join()
   th2:join()
   th3:join()
+  db:begin()
   db:exec("select * from t order by i"):emit()
+  db:commit()
 end}$$
```

Comment:
In this test case there is a transaction that starts after the explicit commit and finishes when `main` returns. When this transaction runs in snapshot isolation, the old and new snapshot implementations behave differently:

- The old snapshot implementation starts its snapshot when the select statement runs, so it sees the results of the transactions executed by the dispatched threads.

- The new snapshot implementation (modsnap) starts its snapshot when the transaction begins, so it does not see the results of the transactions executed by the dispatched threads.

The changes in this PR wrap the select statement in an explicit transaction so that the two snapshot implementations produce the same output.